docs/source-no-noun-replacement-params.md

```diff
@@ -0,0 +1,69 @@
+# source-no-noun-replacement-params
+
+Ensure that source strings do not contain replacement parameters where nouns, noun
+phrases, or adjectives are substituted in to the string. The reason is that the
+spelling of the nouns and adjectives can vary with the linguistic case of that
+part of the sentence. Also, the spelling of the translations of the definite and
+indefinite articles "a", "an", and "the" varies with the gender and plurality of
+the noun or noun phrase being substituted in.
+
+Examples of bad source strings:
+
+```
+Delete the {fileType}.
+Would you like to create a new {fileType} file?
```

Comment:
IMO this explanation is a bit hard to grasp quickly, meaning it's more likely to be ignored.

I think after providing examples of bad source strings, it's very important to provide a good way to rewrite each of them. For me, it's not immediately obvious what's wrong with the second string nor would I know how to rewrite it.

Additionally, maybe it would make sense to provide examples of why each of them would be hard to translate? Like
> The {messageType} has been sent.
in Polish it's
> (messageType="e-mail") E-mail został wysłany. (masculine) 
> (messageType="direct message") Wiadomość bezpośrednia została wysłana. (feminine)

One more thing: `a new {fileType}` would not be caught by this rule, so I suppose it should not be used as an example.